### DISCUSSION TITLE
Include component files before applying Tailwind

### DISCUSSION THREAD

**Discussion by unscatty** [saadeghi/daisyui#2422] [Ideas / feature requests]
First of all, thank you for daisy, I really love it :heart:

Is it possible to include a version of the component files (`styled`, `unstyled`, `utilities`, etc.) before being transformed by Tailwind?
I'm currently working on a [UnoCSS](https://unocss.dev) version of Daisy, and will find this useful because it allows to reuse applied utility classes in components (via @apply) and let Uno handle modifications by using `theme` option, like modifying breakpoints, spacing, etc.
Currently, I do this by hand, but would be great if this can be imported directly from daisy. Or maybe a different package to keep the package size minimum.

I can submit a PR if needed.

**Comment by saadeghi** [comment]
The source files are on GitHub but not in the Node package.  
However, if you want to have all the source files inside `node_modules` you can install the GitHub repo itself 😁  
```
npm install daisyui-source@https://github.com/saadeghi/daisyui/archive/refs/tags/v3.9.3.tar.gz
```
It will go in `node_modules/daisyui-source` so it won't be mixed with the original `node_modules/daisyui` package.

---

Also, have you seen this project? https://github.com/kidonng/unocss-preset-daisy

**Reply by unscatty** [reply]
Hadn't actually thought about using daisy's source :sweat_smile:, that's a good one.

I'm aware of the existing preset for daisy, actually I used it as a base. However, I'm building this new preset specifically to support using modifiers like `hover`, `focus`, `sm`, etc. It is available at https://github.com/unscatty/unocss-preset-daisy if want to take a quick look :wink:

**Reply by saadeghi** [reply]
Sounds good. Let me know when it's ready to use, so I will add a link on daisyUI website's footer.

**Reply by unscatty** [reply]
That'd be awesome! Latest version fully works, I haven't thoroughly tested every component, but seems everything's right. There are some minor issues with component z-positioning, but are Uno specific, and can easily be fixed.
But, yeah, latest version works as expected :+1:
